lib/index.js

```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
+   */
+  getPlaceholder: function(){
+    return this.options.placeholder;
+  },
+
+  /**
+   * Set the value of the input element's placeholder
+   * @param {String} placeholder the text to use as the input element's placeholder
+   * @returns {MapboxGeocoder} this
+   */
+  setPlaceholder: function(placeholder){
+    this.placeholder = (placeholder) ? placeholder : this._getPlaceholderText();
+    this._inputEl.placeholder = this.placeholder;
+    return this
+  },
+
+  /**
+   * Gets the bounding box used by the plugin
+   * @returns {Array<Number>} the bounding box, if any
+   */
+  getBbox: function(){
+    return this.options.bbox;
+  },
+
+  /**
+   * Sets the bounding box to limit search results to
+   * @param {Array<Number>} bbox a bounding box given as an array in the format [minX, minY, maxX, maxY].
+   * @returns {MapboxGeocoder} this
+   */
+  setBbox: function(bbox){
+    this.options.bbox = bbox;
+    return this;
+  },
+
+  /**
+   * Get a list of the countries to limit search results to 
+   * @returns {String} a comma separated list of countries to limit to, if any
+   */
+  getCountries: function(){
+    return this.options.countries;    
+  },
+
+  /**
+   * Set the countries to limit search results to
+   * @param {String} countries a comma separated list of countries to limit to 
+   * @returns {MapboxGeocoder} this
+   */
+  setCountries: function(countries){
+    this.options.countries = countries;
+    return this;
+  },
+
+  /**
+   * Get a list of the types to limit search results to 
+   * @returns {String} a comma separated list of types to limit to 
+   */
+  getTypes: function(){
+    return this.options.types;    
+  },
+
+  /**
+   * Set the countries to limit search results to
+   * @param {String} countries a comma separated list of countries to limit to 
+   * @returns {MapboxGeocoder} this
+   */
+  setTypes: function(types){
+    this.options.types = types;
+    return this;
+  },
+
+  /**
+   * Get the minimum length used in the plugin
+   * @returns {Number}
+   */
+  getMinLength: function(){
+    return this.options.minLength;
+  },
+
+  /**
+   * Set the minimum length value used by the plugin
+   * @param {Number} minLength the minimum length in characters
+   * @returns {MapboxGeocoder} this
+   */
+  setMinLength: function(minLength){
+    this.options.minLength = minLength;
+    if (this._typeahead)  this._typeahead.minLength = minLength;
+    return this;
+  },
+
+  /**
+   * Get the limit value used by the plugin
+   * @returns {Number} the limit value
+   */
+  getLimit: function(){
+    return this.options.limit;
+  },
+
+  /**
+   * Set the limit value used by the plugin
+   * @param {Number} limit the number of search results to return 
+   * @returns {MapboxGeocoder} 
+   */
+  setLimit: function(limit){
+    this.options.limit = limit;
+    if (this._typeahead) this._typeahead.options.limit = limit;
+    return this;
+  },
+
+  /**
+   * Get the filter function used by the plugin
```

Comment:
I'm not exactly sure what `filter` does in suggestions, maybe we could add a little more to add context?

Reply:
It filters the results shows :wink: See https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-limit-region/